johnsonandjohnson/Bodiless-JS
# fix(migration-tool): ENOTDIR: not a directory error during build

## Changes
* fixed a bug regarding "ENOTDIR: not a directory error" during gatsby build
* enhanced logging of the migration tool scraper

## Notes
* "ENOTDIR: not a directory error" caused by having a resource (file in gatsby static directory) and a page with the same name. This case may happen:
1. migration tool hits pageA. the server responds with 2xx. migration tool creates a page for pageA
2. migration tool hits pageB which is redirected to pageA. the sever responds with 304 (not modified) for pageA. migration tool creates a file for pageA

## Test Instructions

1. Configure test server so that it servers the case which led to the build error
```
git clone git@github.com:beliayeu/migration-tool-test-server.git
cd migration-tool-test-server
git checkout test/page-and-resource-conflict
npm i
npm run start
```

2. Migrate url served by test server configured in 1
3. Build static static

Expected: no build errors
Actual: build fails with an error
